### PR TITLE
Master l10n cl pos pim bmya

### PR DESCRIPTION
**Pull request by Danisan**
Description of the issue/feature this PR addresses:
This is a work-in-progress commit, to be tested and complemented. It fixes the weird behavior of #97417

Current behavior before PR:
the document type 110 (export invoice) was selected by default when the partner was empty (the xml_id was incorrect)
once corrected this, more unexpected behaviors arrived:
- the journal for the pos order should not be a sales journal since when validated without document type raises an error from LatAm document type constraints.

Desired behavior after PR is merged:
when selecting products in the pos and doing the sale, automatically create the default document and post it

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by fvz-odoo** [comment]
Hello @Danisan 

I tried to test but I keep getting a traceback error when validating the orders. Tried with and without a customer, with and without invoice... could you please check?

Thanks!

Start tag expected, '<' not found, line 1, column 1
Traceback (most recent call last):
  File "/data/build/odoo/odoo/http.py", line 1510, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "/data/build/odoo/odoo/service/model.py", line 134, in retrying
    result = func()
  File "/data/build/odoo/odoo/http.py", line 1539, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "/data/build/odoo/odoo/http.py", line 1736, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "/data/build/odoo/addons/website/models/ir_http.py", line 221, in _dispatch
    response = super()._dispatch(endpoint)
  File "/data/build/odoo/odoo/addons/base/models/ir_http.py", line 136, in _dispatch
    result = endpoint(**request.params)
  File "/data/build/odoo/odoo/http.py", line 664, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "/data/build/odoo/addons/web/controllers/dataset.py", line 42, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/data/build/odoo/addons/web/controllers/dataset.py", line 33, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/data/build/odoo/odoo/api.py", line 457, in call_kw
    result = _call_kw_model(method, model, args, kwargs)
  File "/data/build/odoo/odoo/api.py", line 430, in _call_kw_model
    result = method(recs, *args, **kwargs)
  File "/data/build/odoo/addons/pos_sale/models/pos_order.py", line 40, in create_from_ui
    order_ids = super(PosOrder, self).create_from_ui(orders, draft)
  File "/data/build/enterprise/pos_l10n_se/models/pos_blackbox_sweden.py", line 174, in create_from_ui
    return super(PosOrder, self).create_from_ui(regular_orders, draft)
  File "/data/build/odoo/addons/point_of_sale/models/pos_order.py", line 693, in create_from_ui
    order_ids.append(self._process_order(order, draft, existing_order))
  File "/data/build/odoo/addons/point_of_sale/models/pos_order.py", line 148, in _process_order
    pos_order._generate_pos_order_invoice()
  File "/data/build/odoo/addons/point_of_sale/models/pos_order.py", line 640, in _generate_pos_order_invoice
    new_move.sudo().with_company(order.company_id)._post()
  File "/data/build/enterprise/l10n_cl_edi/models/account_move.py", line 142, in _post
    move._l10n_cl_create_dte()
  File "/data/build/enterprise/l10n_cl_edi/models/account_move.py", line 486, in _l10n_cl_create_dte
    signed_dte = self._sign_full_xml(
  File "/data/build/enterprise/l10n_cl_edi/models/l10n_cl_edi_util.py", line 257, in _sign_full_xml
    self._xml_validator(signature, 'sig')
  File "/data/build/enterprise/l10n_cl_edi/models/l10n_cl_edi_util.py", line 224, in _xml_validator
    return tools.validate_xml_from_attachment(self.env, xml_to_validate, xsd_fname, self.env['ir.attachment']._l10n_cl_edi_load_xsd_files)
  File "/data/build/odoo/odoo/tools/xml_utils.py", line 278, in validate_xml_from_attachment
    _check_with_xsd(xml_content, xsd_name, env)
  File "/data/build/odoo/odoo/tools/xml_utils.py", line 56, in _check_with_xsd
    xsd_schema = etree.XMLSchema(etree.parse(stream, parser=parser))
  File "src/lxml/etree.pyx", line 3521, in lxml.etree.parse
  File "src/lxml/parser.pxi", line 1876, in lxml.etree._parseDocument
  File "src/lxml/parser.pxi", line 1896, in lxml.etree._parseMemoryDocument
  File "src/lxml/parser.pxi", line 1784, in lxml.etree._parseDoc
  File "src/lxml/parser.pxi", line 1141, in lxml.etree._BaseParser._parseDoc
  File "src/lxml/parser.pxi", line 615, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 725, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 654, in lxml.etree._raiseParseError
  File "<string>", line 1
lxml.etree.XMLSyntaxError: Start tag expected, '<' not found, line 1, column 1

File: addons/point_of_sale/models/pos_order.py [hunk 1]
```diff
@@ -144,11 +144,14 @@ def _process_order(self, order, draft, existing_order):
             pos_order._create_order_picking()
             pos_order._compute_total_cost_in_real_time()
 
-        if pos_order.to_invoice and pos_order.state == 'paid':
+        if pos_order._should_create_invoice():
             pos_order._generate_pos_order_invoice()
 
         return pos_order.id
 
+    def _should_create_invoice(self):
```
Comment (pimodoo) [inline addons/point_of_sale/models/pos_order.py:152]: I don't think those changes in point of sale are needed, you can just put to_invoice to 'True' in you override of 'def _order_fields(self, ui_order):' like you are overriding the partner_id

File: addons/l10n_cl_pos/views/res_config_settings_views.xml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<odoo>
+    <data>
+
+        <record id="res_config_settings_view_form_inherit" model="ir.ui.view">
+            <field name="name">res.config.settings.pos.cl</field>
+            <field name="model">res.config.settings</field>
+            <field name="inherit_id" ref="point_of_sale.res_config_settings_view_form"/>
+            <field name="priority" eval="8"/>
+            <field name="arch" type="xml">
+                <field name="pos_journal_id" position="after">
```
Comment (pimodoo) [inline addons/l10n_cl_pos/views/res_config_settings_views.xml:11]: I don't think it is used

File: addons/l10n_cl_pos/views/res_config_settings_views.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                    <field name="country_code" invisible="1"/>
+                </field>
+                <field name="pos_journal_id" position="attributes">
```
Comment (pimodoo) [inline addons/l10n_cl_pos/views/res_config_settings_views.xml:14]: Shouldn't this change should be in 'point_of_sale'? As it seems a good idea, and once the localisation CL is installed it is applied for all pos_config of any companies.

**Comment by Danisan** [comment]
closed because was solved by several other PRs

**Pull request closed without merging**